### ISSUE TITLE
Ability to manually create a Contact

### ISSUE THREAD

**Issue by opera-aberglund** [louis-langholtz/PlayRho#555]
### Desired Behavior:
To create a copy of the `World` state, I must be able to copy all the contacts of the previous world to the new one. 

### Actual Behavior:
I am able to get all information about contacts, and I can construct them from the `Contact` constructor, but there is no way to manually add them to `m_contactBuffer` and `m_contacts` in the world state and get a new ContactID.

**Comment by louis-langholtz** [comment]
Hmm... in what ways are contacts not caches? That's the question I need to consider.

In the past, I'd been of the position that they're only caches. With the exception of the ability for users to tweak contacts via functions like `SetContact`. Sigh.

How important are persisting those tweaks for serialization? Presumably, they're important for your use.

I also wonder how exact the reproduction of contacts is when serializing without contact info. The same number of contacts should reproduce IMO. The ordering may be different though.

This brings up another consideration of mine that I've had: aren't contacts conceptually like joints. So if joints creation and destruction can be user managed, why not contacts too? With perhaps a component that automagically manages these - like is done now for contacts.

**Comment by opera-aberglund** [comment]
> How important are persisting those tweaks for serialization? Presumably, they're important for your use.

Yeah that's the deal; anything that allows the serialized world to diverge by an inch is a major issue for me, so I need to be sure it's a true copy. From my perspective it could be developed in either direction: that contacts are purely cached data that will always be automatically recreated the same way in the serialized world, or they contain unique and modifiable information that can always be read and recreated whenever. There's no middle ground though.
